src/prog_models/prognostics_model.py

```diff
@@ -1113,21 +1126,76 @@ def calc_error(self, times: List[float], inputs: List[InputContainer], outputs:
         """
         method = kwargs.get('method', 'MSE')
 
-        # Call appropriate error calculation method
-        if method.lower() == 'mse':
-            return calc_error.MSE(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'max_e':
-            return calc_error.MAX_E(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'rmse':
-            return calc_error.RMSE(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'mae':
-            return calc_error.MAE(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'mape':
-            return calc_error.MAPE(self, times, inputs, outputs, **kwargs)
-
-        # If we get here, method is not supported
-        raise ProgModelInputException(f"Error method '{method}' not supported")
-    
+        method_map = {
+            'mse': calc_error.MSE,
+            'max_e': calc_error.MAX_E,
+            'rmse': calc_error.RMSE,
+            'mae': calc_error.MAE,
+            'mape': calc_error.MAPE
+        }
+
+        try:
+            method = method_map[method.lower()]
+        except KeyError:
+            # If we get here, method is not supported
+            raise ProgModelInputException(f"Error method '{method}' not supported")
```

Comment:
```suggestion
            raise KeyError(f"Error method '{method}' not supported")
```
No longer using the custom exceptions for these.